src/components/LetterInputs/LetterInputs.js

```diff
@@ -1,83 +1,113 @@
   };
 
-  // Getting access to Redux dispatch to send actions and the counter value from the store
-  const dispatch = useDispatch();
-  const counterValue = useSelector((state) => state.increment.value);
- 
   return (
     <>
       <div style={{ display: "flex", alignItems: "center", gap: "1rem" }}>
-        {/* Create an array of input fields based on wordLength. I saw this in a tutorial once, I can't say I know how it's working. sorry!  */}
-        {Array.from({ length: wordLength }, (_, i) => (
+        {letterStates.map((letterState, i) => (
+          // changed to a map to make this clearer for all us beginners
           <input
+            // maybe we'll have some inline styling happening in here for the boxes
             maxLength={1}
             key={i}
-            value={letterStates[i]}
-            // confession: ChatGBT helped me with this below line, I couldn't figure it out on my own!
+            value={letterState.letter}
             onChange={(event) => updateLetterState(i, event.target.value)}
           />
         ))}
       </div>
       {counterValue === 10 ? (
-
-      <Link to='/results'>
-         <button
-        className="submit-word-btn"
-        onClick={() => {
-          const guessedWord = letterStates.join("");
-          console.log(guessedWord);
-          dispatch(update(guessedWord));
-          clearInputFields();
-          dispatch(increment());
-        }}
-      >
-        Submit
-      </button>
-      </Link>
-      ): (
-
-      <button
-        className="submit-word-btn"
-        onClick={() => {
-          const guessedWord = letterStates.join("");
-          console.log(guessedWord);
-          dispatch(update(guessedWord));
-          clearInputFields();
-          dispatch(increment());
-        }}
-      >
-        Submit
-      </button>
+        // ==== UPDATE =====
+        // I think that this actually won't be a 'button' with a link - we'll need to just immediately route to the results page IF the counter value increments to 10
+        // Not sure what that will mean for code but I think below code will change
+        <Link to="/results">
+          <button
+            className="submit-word-btn"
+            onClick={() => {
+              const guessedWord = letterStates.join("");
+              console.log(guessedWord);
+              dispatch(update(guessedWord));
+              dispatch(increment());
+            }}
+          >
+            Submit
+          </button>
+        </Link>
+      ) : (
+        // ===== UPDATE END =====
+        <button
+          className="submit-word-btn"
+          onClick={() => {
+            if (isWordCorrect(letterStates)) {
+              // If all the letters are correct send true to update reducer -> goes to the next word
+              dispatch(update(true));
+              dispatch(increment());
+            }
+            // what happens if false? Nothing? Maybe highlight the letters that have status: false
```

Comment:
Hmm, I think if we have a more lines of logic here, might be a good idea to move it into a separate function and then just called a handleClick function in the button for readability.